rust/worker/src/execution/orchestration/metadata.rs

```diff
@@ -556,13 +563,48 @@ impl MetadataQueryOrchestrator {
         println!("Filtering logs and searching metadata segment");
         self.state = ExecutionState::Filter;
 
-        // TODO: Implement filtering and searching metadata segment
-        // for now we just proxy the items through on the request thread
-        // since in the server we disallow where/where document.
-        // When we implement this we can move it to an operator
+        let metdata_segment = self
+            .get_metadata_segment_from_id(self.sysdb.clone(), &self.metadata_segment_id)
+            .await;
+        let metadata_segment = match metdata_segment {
+            Ok(segment) => segment,
+            Err(e) => {
+                self.terminate_with_error(e, ctx);
+                return;
+            }
+        };
 
-        // TODO: If we were to search the metadata segment we would do it here
-        let filtered_index_offset_ids = None;
+        // TODO create segment in this method instead of constructor
+        let metadata_segment_reader =
+            MetadataSegmentReader::from_segment(&metadata_segment, &self.blockfile_provider).await;
+
+        let filtered_index_offset_ids = match metadata_segment_reader {
+            Ok(reader) => {
+                reader
+                    .query(
+                        self.where_clause.as_ref(),
```

Comment:
Is it worth pulling the query out into an operator right now? AFAIK we won't reuse it but I guess it's not too much work. Thoughts welcome.